--- create_stsgeo_v17a.C.orig	2016-07-11 22:54:27.031366447 +0200
+++ create_stsgeo_v17a.C	2016-07-11 22:52:57.778440615 +0200
@@ -640,7 +640,7 @@
     Double_t posZ = statPos[iStation-1] - stsPosZ;
     //    Double_t posZ = statPos[iStation-1];
     TGeoTranslation* trans = new TGeoTranslation(0., 0., posZ);
-    sts->AddNode(station, iStation, trans);
+    sts->AddNode(station, 1, trans);
     sts->GetShape()->ComputeBBox();
   }
   cout << endl;
