*** create_stsgeo_v18f.C	2017-11-10 13:50:43.850195673 +0100
--- create_stsgeo_v18f_nokeep.C	2017-11-10 13:36:26.853451483 +0100
***************
*** 880,886 ****
  //  TGeoShape* stsShape = new TGeoCompositeShape("stsShape", 
  //                                               "stsBox-stsCone1:trans1-stsCone2:trans2");
  //  TGeoVolume* sts = new TGeoVolume(stsName.Data(), stsShape, gStsMedium);
!   TGeoVolume* sts = new TGeoVolume(stsName.Data(), stsBox, gStsMedium);
  
    // --- Place stations in the STS
    //  for (Int_t iStation = 1; iStation <=8; iStation++) {
--- 880,887 ----
  //  TGeoShape* stsShape = new TGeoCompositeShape("stsShape", 
  //                                               "stsBox-stsCone1:trans1-stsCone2:trans2");
  //  TGeoVolume* sts = new TGeoVolume(stsName.Data(), stsShape, gStsMedium);
! //  TGeoVolume* sts = new TGeoVolume(stsName.Data(), stsBox, gStsMedium);
!   TGeoVolumeAssembly* sts = new TGeoVolumeAssembly(stsName.Data());   // do not produce keeping volumes
  
    // --- Place stations in the STS
    //  for (Int_t iStation = 1; iStation <=8; iStation++) {
***************
*** 2086,2093 ****
  //  
  //  statShape = new TGeoCompositeShape(name, expression.Data());
  //  TGeoVolume* station = new TGeoVolume(name, statShape, gStsMedium);
!   TGeoVolume* station = new TGeoVolume(name, statBox, gStsMedium);
!   
    Double_t subtractedVal;
    
    // --- Place ladders in station
--- 2087,2095 ----
  //  
  //  statShape = new TGeoCompositeShape(name, expression.Data());
  //  TGeoVolume* station = new TGeoVolume(name, statShape, gStsMedium);
! //  TGeoVolume* station = new TGeoVolume(name, statBox, gStsMedium);
!   TGeoVolumeAssembly* station = new TGeoVolumeAssembly(name);   // do not produce keeping volumes
! 
    Double_t subtractedVal;
    
    // --- Place ladders in station
